### DISCUSSION TITLE
分享自己部署成功经验

### DISCUSSION THREAD

**Discussion by oddfar** [oddfar/campus-imaotai#91] [分享部署教程]
使用群辉、威联通、openwrt等，可以在此频道分享自己部署成功的经历，帮助其他人。
- https://github.com/oddfar/campus-imaotai/discussions/categories/show-and-tell

**Comment by mrxxllcc** [comment]
不会,威联通部署失败,求一键docker

**Comment by Qiqia** [comment]
> 

我这边部署的时候是mysql部署失败，看日志是提示没有权限，暂时还不知道怎么解决

**Comment by Qiqia** [comment]
威联通上部署成功，网页可以打开，i茅台账号配置了，但是能不能申购成功还有待测试，分享下注意点吧
1.java8要换成openjdk8，第一次build拉镜像在国内花了很久。
2.nginx的conf文件里面增加一条user root，至于为什么是root，暂时不太清楚。
3.compose.yml文件里面，mysql的配置里面增加  user： xxx:xxx 这里面对应uid和gid，不能用0:0，具体原因不太清楚，因为用0:0就报错没有权限。
4.compoes.yml文件里面，nginx的配置里面增加 user: 0:0 就是root启动，与第二点呼应。
5.在添加i茅台账号的时候，发短信失败，issus里面多说是网的问题，可以在点发送验证码的时候，同时在手机上发送一个验证码。然后用那个验证码登录，暂时不知道有什么隐患。
6.刚配置好，申购还没成功，可能还需要调教，回头再补充。

**Reply by Dalegebai** [reply]
我也是发送短信失败，但是我在本地部署就可以，放到docker运行就会失败。

**Reply by Young9229** [reply]
具体怎么增加，能说说么，小白不知道在哪加

**Comment by TangTree** [comment]
群晖研究很久没搞定，大佬传个docker镜像吧

**Comment by Qiqia** [comment]
> 群晖研究很久没搞定，大佬传个docker镜像吧

没有做镜像，你首先先通过tar命令进行解压，然后每个服务单独启动。查看日志。看看具体报什么错误。
群晖估计和威联通差不多，权限比较纠结。

**Comment by fore3t** [comment]
大佬 有没有碰到领取耐力失败的

**Comment by vip0917** [comment]
user也加了，但是mysql跟nginx马上就意外关闭，mysql日志如下：

2023/09/03 05:56:16 | stderr | 2023-09-03T12:56:16.784912Z 0 [ERROR] Aborting
-- | -- | --
2023/09/03 05:56:16 | stderr | 2023-09-03T12:56:16.784911Z 0 [ERROR] Failed to initialize builtin plugins.
2023/09/03 05:56:16 | stderr | 2023-09-03T12:56:16.784906Z 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2023/09/03 05:56:16 | stderr | 2023-09-03T12:56:16.784886Z 0 [ERROR] Plugin 'InnoDB' init function returned error.
2023/09/03 05:56:16 | stderr | 2023-09-03T12:56:16.184174Z 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
2023/09/03 05:56:16 | stderr | 2023-09-03T12:56:16.184168Z 0 [ERROR] InnoDB: The innodb_system data file 'ibdata1' must be writable
2023/09/03 05:56:16 | stderr | 2023-09-03T12:56:16.184132Z 0 [ERROR] InnoDB: The innodb_system data file 'ibdata1' must be writable

**Comment by vip0917** [comment]
> user也加了，但是mysql跟nginx马上就意外关闭，mysql日志如下：
> 
> 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784912Z 0 [ERROR] Aborting
> 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784911Z 0 [ERROR] Failed to initialize builtin plugins.
> 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784906Z 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
> 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784886Z 0 [ERROR] Plugin 'InnoDB' init function returned error.
> 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.184174Z 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
> 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.184168Z 0 [ERROR] InnoDB: The innodb_system data file 'ibdata1' must be writable
> 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.184132Z 0 [ERROR] InnoDB: The innodb_system data file 'ibdata1' must be writable

user修改为0:0没有错误提示了，但是打开主页，还是如下的提示
Whitelabel Error Page
This application has no explicit mapping for /error, so you are seeing this as a fallback.

Sun Sep 03 21:48:00 CST 2023
There was an unexpected error (type=Not Found, status=404).

**Comment by vip0917** [comment]
> > user也加了，但是mysql跟nginx马上就意外关闭，mysql日志如下：
> > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784912Z 0 [ERROR] Aborting
> > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784911Z 0 [ERROR] Failed to initialize builtin plugins.
> > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784906Z 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
> > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784886Z 0 [ERROR] Plugin 'InnoDB' init function returned error.
> > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.184174Z 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
> > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.184168Z 0 [ERROR] InnoDB: The innodb_system data file 'ibdata1' must be writable
> > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.184132Z 0 [ERROR] InnoDB: The innodb_system data file 'ibdata1' must be writable
> 
> user修改为0:0没有错误提示了，但是打开主页，还是如下的提示 Whitelabel Error Page This application has no explicit mapping for /error, so you are seeing this as a fallback.
> 
> Sun Sep 03 21:48:00 CST 2023 There was an unexpected error (type=Not Found, status=404).

sql改成了自己账户id，组是管理组，yaml正常了，容器全部启动正常，但是打不开主页，报错，无能为力了
Whitelabel Error Page This application has no explicit mapping for /error, so you are seeing this as a fallback.
> 
> Sun Sep 03 21:48:00 CST 2023 There was an unexpected error (type=Not Found, status=404).

**Comment by zhao316678171** [comment]
群辉的一样跑不起来，根据威联通分享的经验，该改的都改了，特别mysql也改了当前组user: 1026:100，但是mysql这个容器跑不起来

**Comment by Qiqia** [comment]
> > > user也加了，但是mysql跟nginx马上就意外关闭，mysql日志如下：
> > > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784912Z 0 [ERROR] Aborting
> > > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784911Z 0 [ERROR] Failed to initialize builtin plugins.
> > > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784906Z 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
> > > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.784886Z 0 [ERROR] Plugin 'InnoDB' init function returned error.
> > > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.184174Z 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
> > > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.184168Z 0 [ERROR] InnoDB: The innodb_system data file 'ibdata1' must be writable
> > > 2023/09/03 05:56:16	stderr	2023-09-03T12:56:16.184132Z 0 [ERROR] InnoDB: The innodb_system data file 'ibdata1' must be writable
> > 
> > 
> > user修改为0:0没有错误提示了，但是打开主页，还是如下的提示 Whitelabel Error Page This application has no explicit mapping for /error, so you are seeing this as a fallback.
> > Sun Sep 03 21:48:00 CST 2023 There was an unexpected error (type=Not Found, status=404).
> 
> sql改成了自己账户id，组是管理组，yaml正常了，容器全部启动正常，但是打不开主页，报错，无能为力了 Whitelabel Error Page This application has no explicit mapping for /error, so you are seeing this as a fallback.
> 
> > Sun Sep 03 21:48:00 CST 2023 There was an unexpected error (type=Not Found, status=404).

不知道你看的什么日志，你要去nginx的那个日志看看，其实这是一个前后端分离的项目，nginx负责前端，server那个负责后端，我们要访问的是nginx那个。

**Comment by Qiqia** [comment]
> 群辉的一样跑不起来，根据威联通分享的经验，该改的都改了，特别mysql也改了当前组user: 1026:100，但是mysql这个容器跑不起来

具体问题得看具体报错。

**Comment by missyun** [comment]
请问下有人遇到这种情况么？
root@alinwei:/volume1/docker/campus-imt-V1.0.7# docker-compose up -d
Starting campus-mysql ...
Starting campus-redis ... error

Starting campus-mysql ... error
ot exists

ERROR: for campus-mysql  Cannot start service campus-mysql: Bind mount failed: '/volume1/docker/campus-imt-V1.0.7/mysql/conf' does not exists

ERROR: for campus-redis  Cannot start service campus-redis: Bind mount failed: '/volume1/docker/campus-imt-V1.0.7/redis/data' does not exists

ERROR: for campus-mysql  Cannot start service campus-mysql: Bind mount failed: '/volume1/docker/campus-imt-V1.0.7/mysql/conf' does not exists
ERROR: Encountered errors while bringing up the project.
root@alinwei:/volume1/docker/campus-imt-V1.0.7#

**Reply by adlofp** [reply]
需要在提示你failed的目录下新建conf文件夹或data文件夹

**Comment by deluo** [comment]
win11的docker部署成功，并且成功预约了一次，分别遇到了java8报错和mysql服务启动失败的问题，都在issue中找到解决方案了，感谢

**Comment by oldshi** [comment]
exsi 安装kubesphere容器集群，部署成功，没有任何问题

**Comment by thocell** [comment]
验证码图片不显示

**Reply by chasays** [reply]
说明服务没有启动成功， 手动看下， 是不是端口冲突之类的？

**Comment by chasays** [comment]
如果预约的时候，提示`InvalidKeyException: Illegal key size`， 需要升级jdk到1.8.0_161 及以上就可以了；。
